atlp-rwanda/Avengers-Phantom-BE
# Ch tests

#### What does this PR do?
This writes and runs tests for our endpoints.
#### Description of Task to be completed?
This PR complete tasks regarding testing on the following endpoints

- end points for CRUD operations for users, roles and Route respectively.

#### How should this be manually tested?
this PR can be tested by (cloning this repo in case you didn't and run `npm i` ), pull this branch and configure  .env refer to example.env file(make sure NODE_ENV=test) then run `npm run test` 
#### Any background context you want to provide?
in this PR I have also modified some endpoints which were not protected and was accessible to anyone those are

- creation, updating and deleting role have to be done by an admin user
-  creation, updating and deleting routes have to be done by an admin/operator user
- only admin is allowed to register user
